Test/Zip.Tests.ps1

```diff
@@ -0,0 +1,571 @@
+
+        $ItemType = 'File'
+    )
+
+    $buildRoot = Get-BuildRoot
+
+    foreach( $item in $Path )
+    {
+        $parent = $item | Split-Path
+        if( $parent )
+        {
+            New-Item -Path (Join-Path -Path $buildRoot -ChildPath $parent) -ItemType 'Directory' -Force -ErrorAction Ignore
+        }
+
+        $destinationPath = Join-Path -Path $buildRoot -ChildPath $item
+        Copy-Item -Path $PSCommandPath -Destination $destinationPath
+    }
+
+    Install-Zip -BuildRoot $buildRoot
+}
+
+function ThenArchiveShouldInclude
+{
+    param(
+        $ArchivePath,
+
+        [Parameter(Position=0)]
+        [string[]]
+        $Path
+    )
+
+    foreach( $item in $Path )
+    {
+        $expectedPath = Join-Path -Path $expandPath -ChildPath $item
+        It ('should include {0}' -f $item) {
+            $expectedPath | Should -Exist
+        }
+    }
+}
+
+function ThenArchiveShouldBeCompressed
+{
+    param(
+        $Path,
+
+        [Int]
+        $GreaterThan,
+
+        [int]
+        $LessThanOrEqualTo
+    )
+
+    $packagePath = Join-Path -Path (Get-BuildRoot) -ChildPath $Path
+    $packageSize = (Get-Item $packagePath).Length
+    Write-Debug -Message ('Package size: {0}' -f $packageSize)
+    if( $GreaterThan )
+    {
+        It ('should have a compressed archive size greater than {0}' -f $GreaterThan) {
+            $packageSize | Should -BeGreaterThan $GreaterThan
+        }
+    }
+
+    if( $LessThanOrEqualTo )
+    {
+        It ('should have a compressed archive size less than or equal to {0}' -f $LessThanOrEqualTo) {
+            $packageSize | Should -Not -BeGreaterThan $LessThanOrEqualTo
+        }
+    }
+
+}
+
+function ThenArchiveShouldNotInclude
+{
+    param(
+        [string[]]
+        $Path
+    )
+
+    foreach( $item in $Path )
+    {
+        It ('package should not include {0}' -f $item) {
+            (Join-Path -Path $expandPath -ChildPath $item) | Should -Not -Exist
+        }
+    }
+}
+
+function ThenTaskFails
+{
+    Param(
+        [String]
+        $error
+    )
+
+    It ('should fail with error message that matches "{0}"' -f $error) {
+        $Global:Error | Should match $error
+    }
+}
+
+function ThenTaskSucceeds
+{
+    It ('should not throw an error message') {
+        $Global:Error | Should BeNullOrEmpty
+    }
+}
+
+function WhenPackaging
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory)]
+        $WithYaml,
+
+        [string]
+        $ToFile
+    )
+
+    # Make sure the build root exists.
+    Get-BuildRoot | Out-Null
+
+    $script:context = $taskContext = New-WhiskeyTestContext -ForBuildRoot 'Repo' -ForBuildServer -ForYaml $WithYaml
+    $taskParameter = $context.Configuration['Build'][0]['Zip']
+
+    $threwException = $false
+    $At = $null
+    
+    $Global:Error.Clear()
+
+    try
+    {
+        Invoke-WhiskeyTask -TaskContext $taskContext -Parameter $taskParameter -Name 'Zip'
+    }
+    catch
+    {
+        $threwException = $true

```

Comment:
Sets `$threwException` in the local function scope, instead of `$script:` scope because it was defined earlier on Line 175.

Also, this variable is never checked in a test assertion in this file.